napari/components/viewer_model.py

```diff
@@ -246,34 +244,30 @@ def active_layer(self, active_layer):
 
         self.events.active_layer(item=self._active_layer)
 
-    def _scene_shape(self):
-        """Get shape of currently viewed dimensions.
-
-        Returns
-        -------
-        centroid : list
-            List of center coordinates of scene, length 2 or 3 if displayed
-            view is 2D or 3D.
-        size : list
-            List of size of scene, length 2 or 3 if displayed view is 2D or 3D.
-        corner : list
-            List of coordinates of top left corner of scene, length 2 or 3 if
-            displayed view is 2D or 3D.
-        """
-        # Scale the camera to the contents in the scene
-        min_shape, max_shape = self._calc_bbox()
-        size = np.subtract(max_shape, min_shape)
-        size = [size[i] for i in self.dims.displayed]
-        corner = [min_shape[i] for i in self.dims.displayed]
+    @property
+    def _sliced_extent_world(self):
+        """(2, D) array: Extent of layers in world coordinates after slicing.
 
-        return size, corner
+        D is either 2 or 3 depending on if the displayed data is 2D or 3D.
+        """
+        if len(self.layers) == 0 and self.dims.ndim != 2:
+            # If no data is present and dims model has not been reset to 0
+            # It means someone has passed more than two axis labels which are
+            # being saved.
```

Comment:
```python
 # It means someone has passed more than two axis labels which are
 # being saved.
```

I am not sure what this means. Can this be clarified?

Reply:
I'd replace "it means" with "this situation can arise when". Is it also possible to end up here by adding and then deleting layers, or would that result in the dims model being reset?

Reply:
have tried to clarify, i think dims model will be reset in that case